tests/test_integration/test_custom_registrations.py

```diff
@@ -0,0 +1,99 @@
+from eth_abi.abi import (
+    decode_single,
+    encode_single,
+)
+from eth_abi.decoding import (
+    BaseDecoder,
+)
+from eth_abi.encoding import (
+    BaseEncoder,
+)
+from eth_abi.exceptions import (
+    DecodingError,
+    EncodingError,
+)
+from eth_abi.registry import (
+    registry,
+)
+
+NULL_ENCODING = b'\x00' * 32
+
+
+def encode_null(x):
+    if x is not None:
+        raise EncodingError('Unsupported value')
+
+    return NULL_ENCODING
+
+
+def decode_null(stream):
+    if stream.read(32) != NULL_ENCODING:
+        raise DecodingError('Not enough data or wrong data')
+
+    return None
+
+
+class EncodeNull(BaseEncoder):
+    word_width = None
+
+    @classmethod
+    def from_type_str(cls, type_str, registry):
+        word_width = int(type_str[4:])
+        return cls(word_width=word_width)
+
+    def encode(self, value):
+        self.validate_value(value)
+        return NULL_ENCODING * self.word_width
+
+    def validate_value(self, value):
+        if value is not None:
+            raise EncodingError('Unsupported value')
+
+
+class DecodeNull(BaseDecoder):
+    word_width = None
+
+    @classmethod
+    def from_type_str(cls, type_str, registry):
+        word_width = int(type_str[4:])
+        return cls(word_width=word_width)
+
+    def decode(self, stream):
+        byts = stream.read(32 * self.word_width)
+        if byts != NULL_ENCODING * self.word_width:
+            raise DecodingError('Not enough data or wrong data')
+
+        return None
+
+
+def test_register_and_use_callables():
+    registry.register('null', encode_null, decode_null)
+
+    assert encode_single('null', None) == NULL_ENCODING
+    assert decode_single('null', NULL_ENCODING) is None
+
+    encoded_tuple = encode_single('(int,null)', (1, None))
+
+    assert encoded_tuple == b'\x00' * 31 + b'\x01' + NULL_ENCODING
+    assert decode_single('(int,null)', encoded_tuple) == (1, None)
+
+    registry.unregister('null')
```

Comment:
This register/unregister being necessary in a test is a hint to me that the API could be improved. The ideal I would look for is the pattern of initializing an object in a fixture, and then not worrying about state from one test polluting another test. That's awkward with this global `registry` singleton.  Something like a `make_registry()` method that initializes a new registry with all the defaults would enables us to do:

```py
@pytest.fixture
def registry():
    return make_registry()
```

Then we don't have to worry about unregistering.

Since there are other more pressing issues, maybe just capture the idea into an issue and we can come back to it later. (if you like it)

Reply:
I thought about making a fixture but just decided it was an okay hack for the time being.  Also, I felt like it would be useful to test the functionality by calling `encode_single` and `decode_single` to make it more like an integration test instead of querying the registry directly and using the result of that.  I'll make an issue for it.

Reply:
Issue created: #77

Reply:
This should be wrapped in a `try/finally`, otherwise it's going to leave the registry polluted when this test fails.

Reply:
PR created for this: #79